src/components/product-switcher/index.tsx

```diff
@@ -0,0 +1,243 @@
+import {
+  KeyboardEventHandler,
+  ReactElement,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from 'react'
+import { useRouter } from 'next/router'
+import classNames from 'classnames'
+import { IconCaret16 } from '@hashicorp/flight-icons/svg-react/caret-16'
+import { Product } from 'common/types'
+import ProductIcon from 'components/icons/product-icon'
+import s from './style.module.css'
+
+const OPTION_LIST_ID = 'product-chooser-option-list'
+const OPTION_ID_PREFIX = 'product-chooser-list-item-'
+
+// TODO: should we put this somewhere else for easier reuse?
```

Comment:
We do have a `config/` folder, which might be an appropriate spot for something like this.

Reply:
Thoughts on putting in a new file like `products.ts`? 🤔

Reply:
Not sure if `@hashicorp/platform-product-meta` is helpful to build off of here? 
<img width="878" alt="CleanShot 2021-12-13 at 12 52 01@2x" src="https://user-images.githubusercontent.com/825855/145863115-fbc4a4dc-e366-42b9-9f51-f2a056021d12.png">

Reply:
The only thing I can think of there is that it doesn't include `"sentinel"`. Should we update that to include it?

Reply:
Oh, also not seeing a HashiCorp Cloud Platform which we need to have in the list of products: https://github.com/hashicorp/web-platform-packages/blob/main/packages/product-meta/index.tsx. I incorrectly assumed that's what the `"hashicorp"` one was for.

Reply:
You can extend it if you'd like, something like: 
<img width="925" alt="CleanShot 2021-12-14 at 09 15 39@2x" src="https://user-images.githubusercontent.com/825855/146015304-6e82747b-8841-403f-b87d-5ba72ae1e657.png">

Reply:
Two updates:

- Used some of this TS beauty @alexcarpenter  shared in `src/common/types.ts` :ts-da:
- Added a `config/products.ts` file (uses TS things from `src/common/types.ts`